src/test/java/it/gov/pagopa/pu/workflow/wf/classification/iuf/classification/IufClassificationWFTest.java

```diff
@@ -0,0 +1,117 @@
+package it.gov.pagopa.pu.workflow.wf.classification.iuf.classification;
+
+import it.gov.pagopa.payhub.activities.activity.classifications.ClearClassifyIufActivity;
+import it.gov.pagopa.payhub.activities.activity.classifications.IufClassificationActivity;
+import it.gov.pagopa.payhub.activities.activity.classifications.TransferClassificationActivity;
+import it.gov.pagopa.payhub.activities.dto.classifications.IufClassificationActivityResult;
+import it.gov.pagopa.payhub.activities.dto.classifications.Transfer2ClassifyDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.config.IufClassificationWfConfig;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.dto.IufClassificationNotifyPaymentsReportingSignalDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.dto.IufClassificationNotifyTreasurySignalDTO;
+import it.gov.pagopa.pu.workflow.wf.classification.iuf.activity.StartTransferClassificationActivity;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.ApplicationContext;
+
+import java.util.Collections;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class IufClassificationWFTest {
+
+  @Mock
+  private ClearClassifyIufActivity clearClassifyIufActivity;
+  @Mock
+  private IufClassificationActivity iufClassificationActivity;
+  @Mock
+  private TransferClassificationActivity transferClassificationActivity;
+  @Mock
+  private StartTransferClassificationActivity startTransferClassificationActivity;
+
+  private IufClassificationWFImpl wf;
+
+
+  @BeforeEach
+  void init() {
+    IufClassificationWfConfig iufClassificationWfConfigMock = Mockito.mock(IufClassificationWfConfig.class);
+    ApplicationContext applicationContextMock = Mockito.mock(ApplicationContext.class);
+
+    Mockito.when(iufClassificationWfConfigMock.buildClearClassifyIufActivityStub())
+      .thenReturn(clearClassifyIufActivity);
+
+    Mockito.when(iufClassificationWfConfigMock.buildIufClassificationActivityStub())
+      .thenReturn(iufClassificationActivity);
+
+    Mockito.when(iufClassificationWfConfigMock.buildStartTransferClassificationActivityStub())
+      .thenReturn(startTransferClassificationActivity);
+
+    Mockito.when(applicationContextMock.getBean(IufClassificationWfConfig.class))
+      .thenReturn(iufClassificationWfConfigMock);
+
+    wf = new IufClassificationWFImpl();
+    wf.setApplicationContext(applicationContextMock);
+
+  }
+
+  @Test
+  void testClassify() {
+    // When
+    wf.classify();
```

Comment:
as sonar is reporting through the coverage, just call the classify method will not produce any results because only invoking the notify* methods the toNotify list is filled.

let's remove the @Test annotation on this method and let's call this method after each testNotify* method